llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -5671,6 +5671,10 @@ class TargetLowering : public TargetLoweringBase {
                                        LoadSDNode *OriginalLoad,
                                        SelectionDAG &DAG) const;
 
+  /// Indicates whether the FRem instruction should be expanded before
+  /// ISel in the LLVM IR.
+  virtual bool shouldExpandFRemInIR() const { return false; };
```

Comment:
Don't introduce a new hook for this. You can check if the operation is going to expand, and then check that there is no call in RuntimeLibcalls (this is essentially how PreISelIntrinsicLowering handles memcpy and co)

Reply:
I see that I can check whether the operation is going to expand using `TargetLowering::isOperationExpand`.  But there is no `RTLIB::Libcall` enum value for the `frem` instruction in `llvm/include/llvm/IR/RuntimeLibcalls.def"` which is what we are trying to expand here; of course, we could include the `fmod` variants here. So, for `frem`, I would just check for  `!TargetLowering::isOperationExpand`, right?

Reply:
I got it now. I needed to map `frem` to the right `fmod` LibCall.